tests/core/data_layer/test_data_rpc.py

```diff
@@ -22,16 +22,22 @@
 from chia.wallet.wallet_node import WalletNode
 from tests.setup_nodes import setup_simulators_and_wallets, bt
 from tests.time_out_assert import time_out_assert
+from tests.util.socket import find_available_listen_port
 from tests.wallet.rl_wallet.test_rl_rpc import is_transaction_confirmed
 
 pytestmark = pytest.mark.data_layer
 nodes = Tuple[WalletNode, FullNodeSimulator]
 
 
-async def init_data_layer(root_path: Path) -> AsyncIterator[DataLayer]:
-    test_rpc_port = uint16(21529)
+async def init_data_layer(root_path: Path, wallet_rpc_port: int) -> AsyncIterator[DataLayer]:
     config = bt.config
-    kwargs = service_kwargs_for_data_layer(root_path, config, test_rpc_port)
+    kwargs = service_kwargs_for_data_layer(
+        root_path=root_path,
```

Comment:
afaict, the `root_path` is just pulled out of `bt` already (line 79), so passing in `bt` might make more sense (which would allow it to be a fixture)

Reply:
`service_kwargs_for_data_layer()` is production code so passing it a blocktools seems unlikely to be proper.  Also, `root_path` and `config` are the standard parameters for other `service_kwargs_for_*()`.